Plot within batch ARI (#211)
AlexsLemonade/sc-data-integration: analysis_templates/01-single-group-integration-check-template.Rmd

Reviewer: Leaving some initial comments here - At first I wondered why we can't just split up the unintegrated SCE object back into it's individual SCEs, rather than having to go back to each individual unintegrated file. My conclusion is this is because our merging _overwrites_ the PCs calculated from individual libraries, and those are the PCAs we want to use in these calculations. Just confirming here!

Reviewer: Also, wanted to say - I think it would be helpful to add _small, short, very brief_ markdown sentence above each of these prep chunks that get hidden in the HTML. Something as quick as "Next, read in the unintegrated SCEs" above a chunk will make it easier to "unhide" given chunks of interest in the HTML.

Author: > Leaving some initial comments here - At first I wondered why we can't just split up the unintegrated SCE object back into it's individual SCEs, rather than having to go back to each individual unintegrated file. My conclusion is this is because our merging overwrites the PCs calculated from individual libraries, and those are the PCAs we want to use in these calculations. Just confirming here!

That is correct. We overwrite them when we calculate the batch corrected PC. Additionally, we subset the genes based on shared genes that are present in the merged object, so we couldn't go back and back-calculate the original PCs either.

```

### Within batch ARI

Here we calculate the what we will refer to as the within bath ARI. 
The goal of this metric is to measure how well a given integration method preserves biological heterogeneity by comparing clustering assignments from pre integration to post integration for each library. 
This function takes each original SCE object, prior to any merging, and performs graph based clustering. 
The same clustering is then performed on the integrated object.

> For each library that is present in the integrated object, the original cluster assignments are compared to the new cluster assignments for those cells post integration by calculating the ARI between the cluster assignments. 

Reviewer: ```suggestion
The original cells' cluster assignments calculated on each individual library in the integrated object are then compared to the cells' new cluster assignments post-integration by comparing ARI between these two cluster assignments. 
```

For each library that is present in the integrated object, the original cluster assignments are compared to the cells' new cluster assignments post integration by calculating the ARI between these two cluster assignments. 
Within-batch ARI values close to 1 indicate that biological heterogeneity has been preserved. 

```{r}
# map across the list of integrated sce objects to compare against the individual objects 
within_batch_ari <- purrr::imap(sce_list, 
                                \(sce, int_method){
                                  calculate_within_batch_ari(individual_sce_list,
                                                             sce,
                                                             seed = params$seed,
                                                             integration_method = int_method,
                                                             batch_column = params$batch_column)
                                }) |>
  # combine results into one dataframe
  dplyr::bind_rows()
```

```{r}
# set order of integration methods on axes
within_batch_ari <- set_integration_order(within_batch_ari)

# plot within batch ARI by integration method

> ggplot(within_batch_ari, aes_string(x = "integration_method_factor" , y = "ari")) +

Reviewer 2: > * for example:
> 
> ```
> a_real_function <- function(numeric_iris_column) {
>   ggplot(iris, aes(x = {{numeric_iris_column}})  + geom_histogram()
> }
> 
> # Usage is **without quotes**
> a_real_function(Sepal.Length)
> ```

I assume you meant something more like:

```
a_real_function <- function(data, numeric_column) {
  ggplot(data, aes(x = {{numeric_column}})  + geom_histogram()
}

a_real_function(iris, Sepal.Length)
```

Because I _know_ you would't use external variables in a function.

> ggplot(within_batch_ari, aes_string(x = "integration_method_factor" , y = "ari")) +

Reviewer: I want everyone to know, and this is true, that I wrote that example intentionally without `iris` to see if it would elicit this exact comment, so this is an all-around team success 🎉 🎉 

Of course, I would _never_ do such a thing in production 😄
